src/runtime/virtcontainers/mount.go

```diff
@@ -349,13 +349,17 @@ func isSymlink(path string) bool {
 	return stat.Mode()&os.ModeSymlink != 0
 }
 
-func bindUnmountContainerRootfs(ctx context.Context, sharedDir, cID string) error {
+func bindUnmountContainerRootfs(ctx context.Context, sharedDir string, con *Container) error {
 	span, _ := trace(ctx, "bindUnmountContainerRootfs")
 	defer span.End()
 
-	rootfsDest := filepath.Join(sharedDir, cID, rootfsDir)
-	if isSymlink(filepath.Join(sharedDir, cID)) || isSymlink(rootfsDest) {
-		mountLogger().WithField("container", cID).Warnf("container dir is a symlink, malicious guest?")
+	if con.state.Fstype != "" && con.state.BlockDeviceID != "" {
```

Comment:
Could you add a comment explaining what the test does. It is explicit from your commit message, but hard to deduce from the code. Or alternatively, add a log / debug message explaining why you exit there.